sklearn/preprocessing/data.py

```diff
@@ -581,7 +582,7 @@ def _reset(self):
             del self.mean_
             del self.var_
 
-    def fit(self, X, y=None):
+    def fit(self, X, y=None, accepting_nan=True):
```

Comment:
I don't think that we allow to change the signature of `fit`.
In #10437, I added an argument to the class `missing_values` to accept NaN but also integer.
However, since that this transformer return a floating X, replacing the integer by NaN after having converting the input should be lead to the same underneath processing.
I think that you should go in that direction.

Reply:
@glemaitre should I add 2 class arguments :  `missing_values` and `ignore_nan` or should I add only one of them?

Reply:
@glemaitre, your comment was unclear. Are you suggesting that we support non-nan missing values as input here, or not?

I think for preprocessing that outputs floats, we might as well just support NaN missing values, at least for now.

I also don't think we need an `ignore_nan` param. I consider it fairly safe to have NaN supported by default. Others might disagree, though.

Reply:
@jnothman if we do not use `ignore_nan` than any preprocessor class using functions like `extmath.py` or `csr/csc_mean_var_axis0` will automatically ignore nan.

So as of now I am not using `ignore_nan` but for later stages if others want `ignore_nan` I would add it.

Reply:
> @glemaitre, your comment was unclear. Are you suggesting that we support non-nan missing values as input here, or not?

Sorry it was 8 days ago before the discussion in the issue.
I agree with you and @ogrisel to go for nan support only for the moment. So we don't need any keyword for the class.

> I also don't think we need an ignore_nan param.

@jnothman 
`_incremental_mean_and_var` is also used in incremental PCA. There we don't want to deal with nan, therefore we will need the parameter I think, isn't it?

Reply:
> _incremental_mean_and_var is also used in incremental PCA. There we don't want to deal with nan, therefore we will need the parameter I think, isn't it?

Actually, I think that you are right that the computation ignoring nan will also work the general case and we don't have really a checking at that point. My only concern would be about a regression performance since that we will check if there is nan value knowing that there is none due to earlier call to check_array. Would it have a significant impact?